src/extension.ts

```diff
@@ -184,9 +184,12 @@ export function activate(context: vscode.ExtensionContext) {
     */
 
     vscode.commands.registerCommand("tsp.openTerminalIP", connectCmd)
-    vscode.commands.registerCommand("InstrumentsExplorer.rename", startRename)
+    vscode.commands.registerCommand("InstrumentsExplorer.rename", async (e) => {
+        await startRename(e)
+    })
```

Comment:
We need an asynchronous rename command because we need to await when settings.json is being updated. If it is synchronous, it may or may not update, that behavior seems to be intermittent.

Reply:
Right, but I thought startRename was declared as an asynchronous function.